### DISCUSSION TITLE
local PDB files -> graphein graph object -> pytorch_geometric Data object including features

### DISCUSSION THREAD

**Discussion by l-Dr-MR-l** [a-r-j/graphein#379] [Q&A]
Hi I'm trying to create a dataloader from local PDB files using graphein.
I'm using `graphein.protein.config.ProteinGraphConfig` and `construct_graph(config=self.graphein_config, path=file_path)` from `graphein.protein.graphs` to generate a graph with node and edge features:
'Graph named '6r49_protein_processed' with 2099 nodes and 2153 edges'

I then use `GraphFormatConvertor(src_format="nx", dst_format="pyg")` from `graphein.ml.conversion` to transform this into a pyg Data object, but at this point it seems to lose all features leaving only coordinates:
'Data(edge_index=[2, 4306], node_id=[2099], coords=[2099, 3], name='6r49_protein_processed', num_nodes=2099)'

Is there a correct way to get what I'm trying to achieve?
Are there any examples I might have missed?

```
ProteinGraphConfig(granularity='atom',
                                                  deprotonate=True,  
                                                  edge_construction_functions= [
                                                      add_peptide_bonds,
                                                      add_atomic_edges,                   # Covalent bonds
                                                      add_ring_status,                    # Needs other edges to add to. Adds ring structures.
                                                      add_bond_order,                     # Needs other edges to add to. Adds single/double bond.
                                                      add_disulfide_interactions,
                                                  ],
                                                  node_metadata_functions= [
                                                                            amino_acid_one_hot,
                                                                            expasy_protein_scale,
                                                                            hydrogen_bond_acceptor,
                                                                            hydrogen_bond_donor,
                                                                            ]
                                                  )
```

Edit:
I've gotten a little bit further using
```
columns = ["edge_index", "node_id", "coords", "atom_type"]
convertor = GraphFormatConvertor(src_format="nx", dst_format="pyg", verbose="all_info", columns=columns)
```
Which now gives a lets me construct a DataLoader with 
`dataloader = DataLoader(graph_list, batch_size=1, shuffle=True, drop_last=False)`

But the output seems off:
```
        for batch in dataloader:
            print("Batch contents:")
            print(f"num edge feats: {batch.num_edge_features}") 
            for key, value in batch.__dict__.items():
                print(f"{key}: {value}")
            print("\n")
```
```
num edge feats: 0
_edge_attr_cls: <class 'torch_geometric.data.data.DataEdgeAttr'>
_tensor_attr_cls: <class 'torch_geometric.data.data.DataTensorAttr'>
_store: {'edge_index': tensor([[   0,  #etc.
```
It doesn't look like edge and node attributes are properly defined, instead all thrown into the `_store`.
I'm just feeling my way through the dark here, again: any example code would be great in speeding me through. (or if I'm going completely in the wrong direction, it would help to know)

**Comment by a-r-j** [comment]
Hi @l-Dr-MR-l, apologies for not catching this earlier! I saw you closed the discussion, did you manage to resolve your issue?

**Comment by jaraujo09** [comment]
Hi @l-Dr-MR-l. I'm having the same problem. Did you manage to resolve the issue? I would like to hear from you or @a-r-j about this.
Many thanks!
